sw/device/tests/rstmgr_sw_rst_ctrl_test.c

```diff
@@ -0,0 +1,353 @@
+        {
+            .name = "RSTMGR",
+            .base = TOP_EARLGREY_RSTMGR_AON_BASE_ADDR,
+            .init = rstmgr_init,
+        },
+    [SPI_DEVICE] =
+        {
+            .name = "SPI_DEVICE",
+            .base = TOP_EARLGREY_SPI_DEVICE_BASE_ADDR,
+            .offset = SPI_DEVICE_CFG_REG_OFFSET,
+            .init = spi_dev_init,
+            .config = spi_dev_config,
+            .prgm_val = 0x3f0c,
+        },
+    [SPI_HOST0] =
+        {
+            .name = "SPI_HOST0",
+            .base = TOP_EARLGREY_SPI_HOST0_BASE_ADDR,
+            .offset = SPI_HOST_CONFIGOPTS_REG_OFFSET,
+            .init = spi_host0_init,
+            .config = spi_host0_config,
+            .prgm_val = 0x3210000,
+        },
+    [SPI_HOST1] =
+        {
+            .name = "SPI_HOST1",
+            .base = TOP_EARLGREY_SPI_HOST1_BASE_ADDR,
+            .offset = SPI_HOST_CONFIGOPTS_REG_OFFSET,
+            .init = spi_host1_init,
+            .config = spi_host1_config,
+            .prgm_val = 0x6540000,
+        },
+    [USB] =
+        {
+            .name = "USB",
+            .base = TOP_EARLGREY_USBDEV_BASE_ADDR,
+            .offset = USBDEV_EP_OUT_ENABLE_REG_OFFSET,
+            .init = usbdev_init,
+            .config = usbdev_config,
+            .prgm_val = 0xc3,
+        },
+    [USBIF] =
+        {
+            .name = "USBIF",
+            .base = TOP_EARLGREY_USBDEV_BASE_ADDR,
+            .offset = USBDEV_RXENABLE_OUT_REG_OFFSET,
+            .init = usbif_init,
+            .config = usbif_config,
+            .prgm_val = 0x695,
+        },
+    [I2C0] =
+        {
+            .name = "I2C0",
+            .base = TOP_EARLGREY_I2C0_BASE_ADDR,
+            .offset = I2C_TIMING0_REG_OFFSET,
+            .init = i2c0_init,
+            .config = i2c0_config,
+            .prgm_val = 0x8b00cfe,
+        },
+    [I2C1] =
+        {
+            .name = "I2C1",
+            .base = TOP_EARLGREY_I2C1_BASE_ADDR,
+            .offset = I2C_TIMING1_REG_OFFSET,
+            .init = i2c1_init,
+            .config = i2c1_config,
+            .prgm_val = 0x114010d8,
+        },
+    [I2C2] = {.name = "I2C2",
+              .base = TOP_EARLGREY_I2C2_BASE_ADDR,
+              .offset = I2C_TIMING2_REG_OFFSET,
+              .init = i2c2_init,
+              .config = i2c2_config,
+              .prgm_val = 0x19ec1595},
+};
+
+bool test_main(void) {
+  // reset all devices
+  for (int n = RSTMGR; n < NUM_NODES; ++n) {
+    test_node[n].init(&test_node[n]);
+  }
+
+  // program test reigster in each device
+  for (int n = SPI_DEVICE; n < NUM_NODES; ++n) {
+    test_node[n].config(&test_node[n]);
+    uint32_t obs_val = read_test_reg(&test_node[n]);
+    if (obs_val == test_node[n].prgm_val) {
+      LOG_INFO("after configure, %s test reg value match!!", test_node[n].name);
+    } else {
+      LOG_ERROR("CHECK-fail:after configure: %s obs != exp : 0x%x != 0x%x",
+                test_node[n].name, obs_val, test_node[n].prgm_val);
+    }
```

Comment:
I think you may be better off using the `CHECK(...)` macro here, and only logging an error message.

Reply:
fxd